doc/expressionLanguage.md

```diff
@@ -301,3 +301,88 @@ two possible types of expressions: Integer (arithmetic operations) or Strings.
 -   update (of type "Boolean"): true -> ${@update * 20} -> ${ 1 \* 20 } -> $ { 20 } -> $ { "20"} -> False
 -   update (of type "Boolean"): false -> ${@update * 20} -> ${ 0 \* 20 } -> $ { 0 } -> $ { "0"} -> False
 -   update (of type "Boolean"): true -> ${trim(@updated)} -> ${trim("true")} -> $ { "true" } -> $ { "true"} -> True
+
+## JEXL Based Transformations
+
+As an alternative, the IoTAgent Library supports as well [JEXL](https://github.com/TomFrost/jexl).
+To use JEXL, you will need to either configure it as default
+language using the `defaultExpressionLanguage` field to `jexl` or configuring
+the usage of JEXL as expression language for a given device:
+
+```
+{
+   "devices":[
+      {
+         "device_id":"45",
+         "protocol":"GENERIC_PROTO",
+         "entity_name":"WasteContainer:WC45",
+         "entity_type":"WasteContainer",
+         "expressionLanguage": "jexl",
+         "attributes":[
+            {
+               "name":"location",
+               "type":"geo:point",
+               "expression": "..."
+            },
+            {
+               "name":"fillingLevel",
+               "type":"Number",
+               "expression": "..."
+            }
+         ]
+      }
+   ]
+}
+```
+
+In the following we provide examples of using JEXL to apply transformations.
+
+### Quick comparison to default language
+
+* JEXL supports the following types: Boolean, String, Number, Object, Array.
+* JEXL allows to navigate and [filter](https://github.com/TomFrost/jexl#collections) objects and arrays.
+* JEXL supports if..then...else... via [ternary operator](https://github.com/TomFrost/jexl#ternary-operator).
+* JEXL additionally supports the following operations:
+  Divide and floor `//`, Modulus `%`, Logical AND `&&` and 
+  Logical OR `||`. Negation operator is `!`
+* JEXL supports [comparisons](https://github.com/TomFrost/jexl#comparisons).
+
+
+For more details, check JEXL language details
+[here](https://github.com/TomFrost/jexl#all-the-details).
+
+### Examples of expressions
+
+The following table shows expressions and their expected outcomes for a measure
+with four attributes: "value" with value 6, "name" with value "DevId629",
+an object with value {name: "John", surname: "Doe"} and array of value [1,3].
```

Comment:
Literals `value`, `name`, `object` and `array` literals refers to measure names?

Maybe obvious :), but I'd like to clarify it...

Reply:
yes

Reply:
I'd suggest the following wording, trying to be more precise:

> The following table shows expressions and their expected outcomes taking into account the following measures at southbound interface:
> * `value` with value 6 (number)
> * `name` with value `"DevId629"` (string)
> * `object` with value `{name: "John", surname: "Doe"} (JSON object)
> * `array` with value `[1, 3]` (JSON Array)

This is because "measure of four attributes" can be a bit confusing. I understand you refer to a "multi-measure composed of four individual measures" but, anyway, I think the wording above is simpler (doesn't need to introduce the "multi-measure" concept) and fully precise.

Reply:
It seems to be also fixed

Reply:
fix by 042c9fd